gson/src/test/java/com/google/gson/functional/PrimitiveTest.java

```diff
@@ -532,159 +528,167 @@ private static String extractElementFromArray(String json) {
 
   @Test
   public void testDoubleNaNSerializationNotSupportedByDefault() {
-    try {
-      double nan = Double.NaN;
-      gson.toJson(nan);
-      fail("Gson should not accept NaN for serialization");
-    } catch (IllegalArgumentException expected) {
-    }
-    try {
-      gson.toJson(Double.NaN);
-      fail("Gson should not accept NaN for serialization");
-    } catch (IllegalArgumentException expected) {
-    }
+    var e = assertThrows(IllegalArgumentException.class, () -> gson.toJson(Double.NaN));
+    assertThat(e)
+        .hasMessageThat()
+        .isEqualTo(
+            "NaN is not a valid double value as per JSON specification. To override this behavior,"
+                + " use GsonBuilder.serializeSpecialFloatingPointValues() method.");
+
+    e = assertThrows(IllegalArgumentException.class, () -> gson.toJson((Double) Double.NaN));
+    assertThat(e)
+        .hasMessageThat()
+        .isEqualTo(
+            "NaN is not a valid double value as per JSON specification. To override this behavior,"
+                + " use GsonBuilder.serializeSpecialFloatingPointValues() method.");
   }
 
   @Test
   public void testDoubleNaNSerialization() {
     Gson gson = new GsonBuilder().serializeSpecialFloatingPointValues().create();
-    double nan = Double.NaN;
-    assertThat(gson.toJson(nan)).isEqualTo("NaN");
     assertThat(gson.toJson(Double.NaN)).isEqualTo("NaN");
+    assertThat(gson.toJson((Double) Double.NaN)).isEqualTo("NaN");
```

Comment:
As side note: It seems all these `Float` and `Double` tests here were flawed before:
The floating point constants such as `Double.NaN` are of the primitive type (e.g. `double` instead of `Double`) so these tests were previously calling `toJson(...)` with a `double` twice, instead of once with `double` and once with `Double`.

Though even now the code is a bit redundant because there is only `toJson(Object)`, so even if called with primitive `double` it will be implicitly boxed to a `Double`.

Reply:
Yeah, I would be inclined to delete the second one since they are indeed both calling the same method with the same value.

Reply:
I have changed it now to use `Gson.toJson(Object, Class)` to cover both `toJson(..., double.class)` and `toJson(..., Double.class)` (which is equivalent to the existing `toJson(...)`), and same for `float`.